File: src/main/java/com/dh/ondot/member/application/command/CreateAddressCommand.java
package com.dh.ondot.member.application.command;

import com.dh.ondot.member.api.request.OnboardingRequest;

public record CreateAddressCommand(
        String roadAddress,
        Double longitude,
        Double latitude
) {
    public static CreateAddressCommand from(OnboardingRequest request) {
        return new CreateAddressCommand(
                request.roadAddress(),
                request.longitude(),
                request.latitude()
        );
    }
}

File: src/main/kotlin/com/dh/ondot/member/application/command/CreateAddressCommand.kt
package com.dh.ondot.member.application.command

import com.dh.ondot.member.api.request.OnboardingRequest

data class CreateAddressCommand(
    val roadAddress: String,
    val longitude: Double,
    val latitude: Double,
) {
    companion object {
        fun from(request: OnboardingRequest): CreateAddressCommand =
            CreateAddressCommand(
                roadAddress = request.roadAddress,
                longitude = request.longitude,
                latitude = request.latitude,
            )
    }
}
